Provisioning/Vault/audit.tf

```diff
@@ -1,7 +1,17 @@
-resource "vault_audit" "file" {
+resource "vault_audit" "stdout" {
   type = "file"
 
   options = {
     file_path = "stdout"
   }
-}
\ No newline at end of file
+}
+
+resource "vault_audit" "otel" {
+  type  = "socket"
+  path  = "otel_socket"
+
+  options = {
+    address     = "opentelemetry-collector.otlp.svc.cluster.local:54525"
```

Comment:
This communication is not well secured by default (no TLS, it's just tcp). Therefore, we have the Kubernetes/Cilium network policies to at least provide some protection when it comes to the question "who can send data to the OTLP" (only Vault Namespace/Pods is allowed to do so). Again, any cluster operator/admin (given sufficient kubectl logs/edit/exec rights) could easily mess with that communication..